pyopencl/array.py

```diff
@@ -1081,6 +1080,26 @@ def __rdiv__(self, other):
 
     __rtruediv__ = __rdiv__
 
+    def __itruediv__(self, other):
+        common_dtype = _get_common_dtype(self, other, self.queue)
+        if common_dtype is not self.dtype:
+            raise TypeError("Cannot cast division output from {!r} to {!r}"
+                            .format(self.dtype, common_dtype))
+
+        if isinstance(other, Array):
+            self.add_event(
+                self._div(self, self, other))
+        else:
+            if other == 1:
+                return self.copy()
```

Comment:
Why the copy?

Reply:
I overlooked that. It should return of course `self`.